### ISSUE TITLE
Exception on duplicate code insertion

### ISSUE THREAD

**Issue by mxschll** [arietimmerman/laravel-url-shortener#9]
Hello,
If a code is generated which already exists in the database, and it is then inserted into the database, an error is thrown.

``` 
SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '1234' for key 'art_urls_code_unique'  
```

I would suggest checking if the code already exists in the database in the creating() mtehod in [URL.php](https://github.com/arietimmerman/laravel-url-shortener/blob/master/src/URL.php). If so, a new code should be created until it does not already exist.

Perhaps one could also dynamically change the length of the code if too many collisions occur.

**Comment by arietimmerman** [comment]
Excellent idea. Will try to work on it. Or alternatively, I would be happy if you can create a pull request.

**Comment by mxschll** [comment]
I'll try to implement it this week.

**Comment by mxschll** [comment]
I created a pull request, see #10

**Comment by diereysaa** [comment]
Following with this idea, why not to check if the URL already exists on the table?
I've ended having several rows with the same URL, and different codes:

![image](https://user-images.githubusercontent.com/38525542/117146475-86aaef80-adb4-11eb-85f4-f6f6d09618ac.png)

 I don't see any downside to this, even the click tracking will work, since it's the same URL.

Right now I'm doing it this way:
`$shortUrl = URL::where('url', 'https://url-goes-here.com')->first() ?? URLShortener::shorten('https://url-goes-here.com');`

But I guess there would be a better/more effective way to do this if we include it inside the `shorten` method itself

Thanks! :)

**Comment by mxschll** [comment]
@arietimmerman What dou you think about this?

**Comment by arietimmerman** [comment]
Thanks for the pull request @mxschll 
I think something like this could be a solution. But I'm not sure if I like the "brute force" mechanism to check if a code is used and after 2 attempts increase the code length.

Wouldn't it be an idea to use some kind of encryption algorithm, and encrypt an incrementer? So this would look like this.

~~~
1. Check the number of shortened URLs
2. Encrypt the number found
~~~

This should guarantee that the number of database lookups is very limited (only one). And the code runs deterministically.

@diereysaa Depending on the application this could be very useful. In some applications you probably still want unique shortened url's, even though the same url is shorted. bitly always generates a unique url for example.

**Comment by arietimmerman** [comment]
Perhaps something like this could also work as well #11 
Or maybe it should be made configurable. That one could choose to either use a random code - with checks like created by @mxschll - or to generate an always unique code (which looks random, but is not random).

**Comment by mxschll** [comment]
You are right, the brute force approach is not ideal, especially if the database has lots of entries. However, an extremely large number of entries would be required before performance problems should arise. So I believe that this approach is manageable.

I like the approach in #11 and that would be sufficient for me. Depending on the application, however, it could be problematic if the codes are not generated randomly, but rather follow a pattern.

One could possibly adapt #10 in such a way that, depending on the number of entries in the database and the associated probability of a collision, the code length is increased in advance. Thus, one could reduce the database queries in advance and the code would still be random.

**Comment by JensenWD** [comment]
Was the fix ever merged? Still getting that error when it tries to create a code with no characters ''
